core/domain/voiceover_domain.py

```diff
@@ -316,3 +317,100 @@ def create_empty(
+
+    def __init__(
+            self,
+            exp_id: str,
+            content_id_to_voiceovers_mapping: Dict[
+                str, Dict[str, Tuple[str, state_domain.VoiceoverDict]]
+            ]
+    ) -> None:
+        """Construct a ExplorationVoiceArtistsLink domain object.
+
+        Args: 
+            exp_id: str. ID of exploration.
+            content_id_to_voiceovers_mapping: Dict. A dictionary with content
+                IDs as keys and nested dicts as values. Each nested dict
+                contains language codes as keys and a 2-tuple as values.
+                The 2-tuple contains voice artist ID as the first element 
+                and VoiceoverDict as the second element.
+        """
+        self.exp_id = exp_id
+        self.content_id_to_voiceovers_mapping = content_id_to_voiceovers_mapping
+
+    def validate(self) -> None:
+        """Validates attributes of ExplorationVoiceArtistsLink."""
+
+        if not isinstance(self.exp_id, str):
+            raise utils.ValidationError(
+                'Expected exploration id to be string, recieved: %s.'
+                % self.exp_id
+            )
+        if self.exp_id == '':
+            raise utils.ValidationError(
+                'Exploration ID should not be empty.'
+            )
+        if not isinstance(self.content_id_to_voiceovers_mapping, dict):
+            raise utils.ValidationError(
+                'Expected content_id_to_voiceovers_mapping to be Dict'
+                'But recieved: %s' % self.content_id_to_voiceovers_mapping
+            )
+        # Validating content within content_id_to_voiceovers_mapping dict.
+        for content_id, nested_dict in (
+        self.content_id_to_voiceovers_mapping.items()):
+            if not isinstance(content_id, str):
+                raise utils.ValidationError(
+                    'Expected content id to be string, recieved: %s.'
+                    % content_id
+                )
+            if not isinstance(nested_dict, dict):
+                raise utils.ValidationError(
+                    'Expected nested dict to be Dictionary, recieved: %s.'
+                    % nested_dict
+                )
+            # Validating dicts inside each content.
+            for language_code, (artist_id, voiceover_dict) in (
+                nested_dict.items()):
+                if not isinstance(language_code, str):
+                    raise utils.ValidationError(
+                    'Expected language to be string recieved: %s'
+                    % language_code
+                )
+                allowed_language_codes = [language['id'] for language in (
+                constants.SUPPORTED_AUDIO_LANGUAGES)]
+                if language_code not in allowed_language_codes:
+                    raise utils.ValidationError(
+                        'Expected language code to be one from supported '
+                        'audio languages recieved: %s' % language_code
+                    )
```

Comment:
There is a function in utils to do this, let's use that to avoid repetition of same logic.

Reply:
Good catch on using supported audio languages and not just all langauges!